### PR TITLE
fix(table): enable horizontal scroll in table

### PR DESCRIPTION
**Pull request by yinglejia**
Closes IBM/carbon-components-angular#

Enable horizontal scroll in table to match Carbon React implementation. Refer Carbon React implementation: https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/DataTable/Table.js#L34

image

image

#### Changelog

**New**

* {{new thing}}

**Changed**

* {{change thing}}

**Removed**

* {{removed thing}}

### CODE REVIEW

**Comment by Akshat55** [comment]
Holding on merging - as we discuss with @carbon/styles team to add display block to `.bx--data-table-content` class. We can then use @hostbind the class - This would prevent unnecessary elements in template & make it easier to manage.

For the time being, you can add `.bx--data-table-content` class to ibm-table tag with display set to block as follows:

```html
<ibm-table class="bx--data-table-content" style="display: block">
     ....
</ibm-table>
```

**Comment by yinglejia** [comment]
> Holding on merging - as we discuss with @carbon/styles team to add display block to `.bx--data-table-content` class. We can then use @hostbind the class - This would prevent unnecessary elements in template & make it easier to manage.
> 
> For the time being, you can add `.bx--data-table-content` class to ibm-table tag with display set to block as follows:
> 
> ```
> <ibm-table class="bx--data-table-content" style="display: block">
>      ....
> </ibm-table>
> ```

Thanks for the update, @Akshat55. Actually my initial attempt was to use @hostbind the class. I did not end up using that approach since the table component also @hostbind `bx--data-table_inner-container` for sticky header. So we would end up having both `.bx--data-table-content` and `bx--data-table_inner-container` on the host. I don't know Carbon team intends to support this or not. Adding display: block is the easy part.

File: src/table/table.component.ts [hunk 1]
```diff
@@ -173,74 +173,76 @@ import { TableRowSize } from "./table.types";
...
-		ibmTable
-		[sortable]="sortable"
-		[noBorder]="noBorder"
-		[ngClass]="{'bx--data-table--sticky-header': stickyHeader}"
-		[size]="size"
-		[striped]="striped"
-		[skeleton]="skeleton">
-		<thead
-			ibmTableHead
+	<div class="bx--data-table-content">
```
Comment (Akshat55) [inline src/table/table.component.ts:176]:
```suggestion
```

Let's apply the style directly to host & add styles.

File: src/table/table.component.ts [hunk 2]
```diff
@@ -173,74 +173,76 @@ import { TableRowSize } from "./table.types";
...
+					<td class="table_end-indicator">
+						<h5>{{getEndOfDataText() | async}}</h5>
+						<button (click)="scrollToTop($event)" class="btn--secondary-sm">
+							{{getScrollTopText() | async}}
+						</button>
+					</td>
+				</tr>
+			</tfoot>
+		</table>
+	</div>
```
Comment (Akshat55) [inline src/table/table.component.ts:245]:
```suggestion
```

File: src/table/table.component.ts [hunk 3]
```diff
@@ -173,74 +173,76 @@ import { TableRowSize } from "./table.types";
...
 	`
```
Comment (Akshat55) [inline src/table/table.component.ts:246]:
```suggestion
	`,
	styles: [`
		:host {
			display: block;
		}
	`]
```

File: src/table/table.component.ts [hunk 4]
```diff
@@ -173,74 +173,76 @@ import { TableRowSize } from "./table.types";
...
 })
 export class Table implements AfterViewInit, OnDestroy {
```
Comment (Akshat55) [inline src/table/table.component.ts:248]: Where best fits in class, add @HostBinding as seen below. Run npm run lint:fix to ensure the order is corect.

```suggestion
export class Table implements AfterViewInit, OnDestroy {
	@HostBinding("class.bx--data-table-content") tableContent = true;
```

**Review by Akshat55 — CHANGES_REQUESTED** [review]
Hi @yinglejia, Thanks for your contribution & sorry for the long delay!

We have decided to go with leaving the styles in components file in carbon v10 (CCA v4), but for v11 they will be added in @carbon/styles package.

That being said, I have also merged master into the branch so just a friendly reminder,  please git pull.

**Comment by yinglejia** [comment]
New PR created: https://github.com/IBM/carbon-components-angular/pull/2235

**Pull request closed without merging**